test/info.c

```diff
@@ -822,6 +822,31 @@ ODBC_TEST(odbc326)
 }
 
 
+/** ODBC-359 When calling SQLBindCol with a lengthPtr but no targetBuffer to get column data length
+    the buffer got dereferenced without first checking if it is set.
+ */
+ODBC_TEST(odbc359)
+{
+  SQLCHAR* create_table= (SQLCHAR *)"CREATE TABLE `odbc359` (`id` bigint(20) NOT NULL, `msg` mediumtext);";
+  OK_SIMPLE_STMT(Stmt, "DROP TABLE IF EXISTS odbc359");
+  OK_SIMPLE_STMT(Stmt, create_table);
+
+  OK_SIMPLE_STMT(Stmt, "INSERT INTO `odbc359` (`id`, `msg`) VALUES(1, 'Hello there!')");
+  SQLWCHAR id[255];
+  SQLLEN idLen, infoLen;
+
+  OK_SIMPLE_STMTW(Stmt, CW("SELECT id, msg from odbc359"));
+  CHECK_STMT_RC(Stmt, SQLBindCol(Stmt, 1, SQL_C_WCHAR, id, sizeof(id), &idLen));
+
+  // Binding lengthBuffer, but no targetBuffer, type must be SQL_C_WCHAR
+  CHECK_STMT_RC(Stmt, SQLBindCol(Stmt, 2, SQL_C_WCHAR, NULL, 0, &infoLen));
+
+  EXPECT_STMT(Stmt, SQLFetch(Stmt), SQL_SUCCESS);
+
```

Comment:
I think what is missing here, is checking that the returned required length is correct. Since the code was crashing, that was apparently never tested. That is not needed to check the fixed issue, but would make the testcase complete. I'd appreciate if you could add that